Gems/AtomLyIntegration/CommonFeatures/Code/Source/Material/EditorMaterialComponent.cpp

```diff
@@ -385,9 +340,36 @@ namespace AZ
                     OnConfigurationChanged();
                 };
 
+                const char* UnknownSlotName = "<unknown>";
+
+                // If this is the default material assignment ID then it represents the default slot which is not contained in any other group
+                if (slot.m_id == DefaultMaterialAssignmentId)
+                {
+                    slot.m_label = "Default Material";
+                }
+                else
+                {
+                    auto slotIter = modelMaterialSlots.find(slot.m_id.m_materialSlotStableId);
+                    if (slotIter != modelMaterialSlots.end())
+                    {
+                        const Name& displayName = slotIter->second.m_displayName;
+                        slot.m_label = !displayName.IsEmpty() ? displayName.GetStringView() : UnknownSlotName;
+
+                        slot.m_defaultMaterialAsset = slotIter->second.m_defaultMaterialAsset;
+                    }
+                    else
+                    {
+                        slot.m_label = UnknownSlotName;
+                    }
+                }
+
                 // if material is present in controller configuration, assign its data
                 const MaterialAssignment& materialFromController = GetMaterialAssignmentFromMap(config.m_materials, slot.m_id);
-                slot.m_materialAsset = materialFromController.m_materialAsset;
+                if (materialFromController.m_materialAsset != slot.m_defaultMaterialAsset) // Prevents the default material from showing up as a filled-in value in the property field
+                {
+                    slot.m_materialAsset = materialFromController.m_materialAsset;
+                }
+
```

Comment:
If I remove the condition, here is what happens:
1. Add a material component
2. Edit one of the material slots instance data, to override a property of the built-in material.
3. The material override field still appears empty, as it should, because it's using the built-in assignment.
4. Save and close the Editor
5. Run the Editor and open the level again
6. Now the material override field shows the built-in material as if the user manually assigned it, even though they didn't.

![image](https://user-images.githubusercontent.com/55155825/126598185-59a7929d-0389-427b-a868-17cb9a415ed4.png)

Do you recommend some other way to prevent this scenario?

Reply:
I understand now.  This is a side effect of your other change to implicitly allow editing property overrides without assigning a material.

The material component controller has no concept of the default material.  It only creates material instances for material overrides set in the configuration.  It only applies property overrides if there is an instance.  

The editor material component populates the component controller configuration.  If the slot contains property overrides without an explicit material assignment then the default material is implicitly assigned.  That’s left over from when I tried to support the same concept of editing property overrides without assigning a material.

There were multiple iterations of this.
One automatically assigned the “default material” from the model if the user attempted to edit the property overrides without assigning a material.  In this case, the asset field would not be grayed out if properties were overridden because the material was auto assigned.  While this worked and having assignments made it clear that some data had changed, some users were confused by the automatic assignment.

Then we made it explicit with a context menu option to assign the default material and effectively unlock the slot for property editing.  

All that was removed as part of the rev the engine reviews that removed a lot of menu options and requested that every action be explicit. 

I do agree that you should be able to edit material instance property overrides without assigning anything.  I personally think the first implementation I mentioned was the correct way to do it.  It does not exhibit the problems you are mentioning since the material is automatically assigned and visible in the UI

But we still can’t unassign incoming data. The user may have explicitly set the value.  It will result in even more confusion if something they set suddenly disappears.

Reply:
Okay, thanks for the explanation. I will put it back how it was before, with property assignments only working when there is a material override.